### PR TITLE
Added passport authorization support

### PR DESCRIPTION
**Pull request by wtrocki**

### CODE REVIEW

File: apps/cloud/package.json [hunk 1]
```diff
@@ -12,18 +12,30 @@
...
     "@external/extending-store": "1.0.0",
-    "@raincatcher/user": "0.0.1",
     "@raincatcher/message": "0.0.1",
+    "@raincatcher/store": "1.0.0",
+    "@raincatcher/user": "0.0.1",
     "@raincatcher/webapi": "0.0.1",
+    "@types/connect-flash": "0.0.31",
+    "@types/cookie-parser": "^1.3.30",
+    "@types/lodash": "^4.14.64",
+    "@types/passport-local": "^1.0.30",
```
Comment (paolobueno) [inline apps/cloud/package.json]: `@types` deps should be `devDependencies`

File: apps/cloud/src/passportSecurity.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            // TODO handle errors
+            // TODO check password
+            done(null, found);
+        });
+    };
+
+    passport.use(new LocalStrategy(authLookup));
+
+    // Session serialization
+    passport.serializeUser(function (user: User, done: Function) {
```
Comment (paolobueno) [inline apps/cloud/src/passportSecurity.ts]: `Function` is pretty much `(any[]) => any`, doesn't `@types/passport` provide a strongly-typed version of this handler?

File: apps/cloud/src/app.ts [hunk 3]
```diff
@@ -19,11 +23,47 @@ app.use(morgan('dev'));
...
+app.use('/tasksSecured', passport.authenticate('local', { session: false }), taskRoute);
+
+
+app.use(function(req, res, next) {
+    var err = new Error('Not Found');
+    next(err);
+});
+
+if (app.get('env') === 'development') {
+    app.use(function (err: any, req: any, res: any, next: any) {
```
Comment (paolobueno) [inline apps/cloud/src/app.ts]: Express has a type for this: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express-serve-static-core/index.d.ts#L29-L31

Any reason for the `any`s?

**Comment by paolobueno** [comment]
Files are using 4 spaces as indentation, we usually go for 2 spaces.

**Comment by paolobueno** [comment]
sorry, clicked wrong button and closed by mistake :stuck_out_tongue:

**Comment by paolobueno** [comment]
added my suggestions as #12 just to make them a bit clearer

File: apps/cloud/src/app.ts [hunk 4]
```diff
@@ -19,11 +23,47 @@ app.use(morgan('dev'));
...
+            error: err
+        });
+    });
+}
+
+// production error handler
+// no stacktraces leaked to user
+app.use(function (err: any, req: any, res: any, next: any) {
+    res.status(err.status || 500);
+    res.render('error', {
```
Comment (paolobueno) [inline apps/cloud/src/app.ts]: The cloud app is a strict JSON api, right? We should be sending error metadata in json instead of trying to `render` a view

**Comment by wtrocki** [comment]
@paolobueno Thanks for review. I will leave this PR open to continue investigations.

ping @TommyJ1994

**Comment by wtrocki** [comment]
@paolobueno Merging back - Going to investigate both session clients and api key aproach in different PR's ( need that changes in master to verify swagger work)

**Pull request merged**